merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/MissionModelProcessor.java

```diff
@@ -80,22 +82,48 @@ public boolean process(final Set<? extends TypeElement> annotations, final Round
     for (final var element : roundEnv.getElementsAnnotatedWith(MissionModel.class)) {
       final var packageElement = (PackageElement) element;
       try {
-        final var missionModelRecord = missionModelParser.parseMissionModel(packageElement);
+        final var missionModelRecord$ = missionModelParser.parseMissionModel(packageElement);
 
         final var generatedFiles = new ArrayList<>(List.of(
-            missionModelGen.generateMerlinPlugin(missionModelRecord),
-            missionModelGen.generateSchedulerPlugin(missionModelRecord)));
+            missionModelGen.generateMerlinPlugin(missionModelRecord$),
+            missionModelGen.generateSchedulerPlugin(missionModelRecord$)));
 
-        missionModelRecord.modelConfigurationType
-            .flatMap(configType -> missionModelGen.generateMissionModelConfigurationMapper(missionModelRecord, configType))
+        missionModelRecord$.modelConfigurationType
+            .flatMap(configType -> missionModelGen.generateMissionModelConfigurationMapper(missionModelRecord$, configType))
             .ifPresent(generatedFiles::add);
 
         generatedFiles.addAll(List.of(
-            missionModelGen.generateMissionModelFactory(missionModelRecord),
-            missionModelGen.generateSchedulerModel(missionModelRecord),
-            missionModelGen.generateActivityActions(missionModelRecord),
-            missionModelGen.generateActivityTypes(missionModelRecord)
+            missionModelGen.generateMissionModelFactory(missionModelRecord$),
+            missionModelGen.generateSchedulerModel(missionModelRecord$),
+            missionModelGen.generateActivityActions(missionModelRecord$),
+            missionModelGen.generateActivityTypes(missionModelRecord$)
         ));
+
+        final var autoValueMapperRequests = roundEnv.getElementsAnnotatedWith(AutoValueMapper.class).stream().map(Element::asType).toList();
+        {
+          final var recordType = elementUtils.getTypeElement("java.lang.Record").asType();
+          for (final var request : autoValueMapperRequests) {
+            if (!typeUtils.isSubtype(request, recordType)) {
```

Comment:
Rather than using subtype analysis, it's better to ask the source Element if it is of kind `ElementKind.RECORD`. This check is both cheaper and clearer, and it avoids accessing elements that might not be accessible from the root element (which is generally important for incremental compilation support in e.g. Gradle.)
```suggestion
            if (request.getKind() == ElementKind.RECORD) {
```
You'll need to defer the `Element::asType` transformation you currently have on these elements until later, because `getKind()` is a method on `Element`, not on `TypeMirror`. (Generally, I've found it's more convenient to work with source `Element`s up until the point you really need to convert it to a prototypical `TypeMirror`.

Reply:
Change made :+1: (Though I'm still fuzzy on `Element` vs `TypeElement`, but both of them have a `getKind()` method, so 🤷 )